### ISSUE TITLE
RS0042 incorrectly firing for ref usage

### ISSUE THREAD

**Issue by jaredpar** [dotnet/roslyn-analyzers#7128]
Consider the [following code](https://github.com/dotnet/roslyn/blob/4fff146d61a23b7e231a1c7a532692e2e6e88e96/src/Compilers/Core/Portable/Collections/TemporaryArrayExtensions.cs#L37-L38
) in roslyn

```csharp
        public static ref TemporaryArray<T> AsRef<T>(this in TemporaryArray<T> array)
            => ref Unsafe.AsRef(in array);
```

This code is not copying a `TemporaryArray<T>` but just downgrading a `in` to a `ref`. This is firing RS0042 which seems incorrect though because there is no copy of the value here, just a change of the ref kind.

**Comment by sharwell** [comment]
That helper method exists for the sole purpose of avoiding diagnostics. Can you provide a more complete example showing how the error was observed?

**Comment by jaredpar** [comment]
The error was fired for that exact method. The `ref Unsafe.AsRef(in array)` is where it fires. You can see in this PR where I had to suppress RS0042 at this location 

https://github.com/dotnet/roslyn/pull/71507/files#diff-0e7827689d069ca296c5c6ccf49bbf565e5eb4e6700b3ddd7769916a7b81b1e3

**Comment by sharwell** [comment]
Weird. So far I have not been able to reproduce this.

**Comment by sharwell** [comment]
Looks like #89736 changed the signature of Unsafe.AsRef<T> to a form that isn't supported by the analyzer. Should be an easy fix.

**Comment by jaredpar** [comment]
That explains the timing of why I just started getting the analyzer failure.
